topic-operator/src/main/java/io/strimzi/operator/topic/Session.java

```diff
@@ -229,11 +215,29 @@ public void handle(Long oldTimerId) {
                     }
                 };
                 periodic.handle(null);
-                promise.future().onComplete(start);
                 LOGGER.info("Started");
             });
     }
 
+    Future<Void> startWatcher(boolean firstTime) {
+        Promise<Void> promise = Promise.promise();
+        try {
+            LOGGER.debug("Watching KafkaTopics matching {}", config.get(Config.LABELS).labels());
+
+            Session.this.topicWatch = kubeClient.customResources(CustomResourceDefinitionContext.fromCrd(Crds.kafkaTopic()), KafkaTopic.class, KafkaTopicList.class, DoneableKafkaTopic.class)
+                    .inNamespace(config.get(Config.NAMESPACE)).withLabels(config.get(Config.LABELS).labels()).watch(watcher);
+            LOGGER.debug("Watching setup");
+            if (firstTime) {
+                // start the HTTP server for healthchecks
+                healthServer = this.startHealthServer();
```

Comment:
There's no need for this method to start the health server. At the call site you currently do: 

```java
 startWatcher(true).onComplete(start);
```
You do easily make that

```java
 startWatcher(true).compose(
    ignored -> {
      Session.this.healthServer = startHealthServer()
      return Future.succeededFuture();
    }).onComplete(start);
```